bpf/bpf_lxc.c

```diff
@@ -1283,9 +1292,19 @@ static __always_inline int handle_ipv4_from_lxc(struct __ctx_buff *ctx, __u32 *d
 #endif /* TUNNEL_MODE */
 
 	if (is_defined(ENABLE_HOST_ROUTING)) {
+#ifdef THIS_INTERFACE_PARENT_IFINDEX
+		int oif = THIS_INTERFACE_PARENT_IFINDEX;
+#else
 		int oif = 0;
+#endif
+		if (oif > 0 && ct_is_reply4(get_ct_map4(tuple), tuple)) {
+			ret = ipv4_l3(ctx, ETH_HLEN, NULL, NULL, ip4);
+			if (ret == CTX_ACT_OK)
+				ret = redirect_neigh(oif, NULL, 0, 0);
+		} else {
+			ret = fib_redirect_v4(ctx, ETH_HLEN, ip4, false, false, ext_err, &oif);
+		}
 
-		ret = fib_redirect_v4(ctx, ETH_HLEN, ip4, false, false, ext_err, &oif);
```

Comment:
Curious, have you looked at what changes are needed to re-use `fib_redirect_v4()` here? I'd be optimistic that it's possible.